framework/src/base/FEProblem.C

```diff
@@ -1468,10 +1468,15 @@ void
 FEProblem::addAuxKernel(const std::string & kernel_name, const std::string & name, InputParameters parameters)
 {
   parameters.set<FEProblem *>("_fe_problem") = this;
+
+  if (_displaced_problem != NULL)
```

Comment:
In `AuxiliarySystem::addKernel`, a new parameter `_aux_sys` is inserted, which is always the undisplaced system. `_sys` in `AuxKernel` is actually never used before. If we did not change here, it will remain unused because I will need to grab the displaced variable direction from the displaced problem.